File: software/examples/perl/example_simple.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletLaserRangeFinderV2;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Laser Range Finder Bricklet 2.0

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $lrf = Tinkerforge::BrickletLaserRangeFinderV2->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Turn laser on and wait 250ms for very first measurement to be ready
$lrf->set_enable(1);
select(undef, undef, undef, 0.25);

# Get current distance
my $distance = $lrf->get_distance();
print "Distance: $distance cm\n";

print "Press key to exit\n";
<STDIN>;

# Turn laser off
$lrf->set_enable(0);

$ipcon->disconnect();

File: software/examples/ruby/example_simple.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_laser_range_finder_v2'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Laser Range Finder Bricklet 2.0

ipcon = IPConnection.new # Create IP connection
lrf = BrickletLaserRangeFinderV2.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Turn laser on and wait 250ms for very first measurement to be ready
lrf.set_enable true
sleep 0.25

# Get current distance
distance = lrf.get_distance
puts "Distance: #{distance} cm"

puts 'Press key to exit'
$stdin.gets

# Turn laser off
lrf.set_enable false

ipcon.disconnect
